pkg/providers/cf/provider.go

```diff
@@ -0,0 +1,363 @@
+	services, err := p.MarshalUnmarshal[dTypes.Services](cfApp.Services)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	routeSpec := p.ParseRouteSpec(cfApp.Routes, cfApp.RandomRoute, cfApp.NoRoute)
+	docker, err := p.MarshalUnmarshal[dTypes.Docker](cfApp.Docker)
+
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	sidecars, err := p.MarshalUnmarshal[dTypes.Sidecars](cfApp.Sidecars)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	processes, err := p.MarshalUnmarshal[dTypes.Processes](cfApp.Processes)
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	var labels, annotations map[string]*string
+
+	if cfApp.Metadata != nil {
+		labels = cfApp.Metadata.Labels
+		annotations = cfApp.Metadata.Annotations
+	}
+	appManifestProcess, inlineProcess, err := p.ParseProcessSpecs(cfApp)
+
+	if err != nil {
+		return dTypes.Application{}, err
+	}
+	var appManifestProcessTemplate *dTypes.ProcessSpecTemplate
+
+	if appManifestProcess != nil {
+		appManifestProcessTemplate = appManifestProcess
+	}
+	if inlineProcess != (nil) {
+		processes = append(processes, *inlineProcess)
+	}
+	app := dTypes.Application{
+		Metadata: dTypes.Metadata{
+			Name:        cfApp.Name,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		Timeout:    timeout,
+		BuildPacks: cfApp.Buildpacks,
+		Env:        cfApp.Env,
+		Stack:      cfApp.Stack,
+		Services:   services,
+		Routes:     routeSpec,
+		Docker:     docker,
+		Sidecars:   sidecars,
+		Processes:  processes,
+	}
+	if appManifestProcessTemplate != nil {
+		app.ProcessSpecTemplate = *appManifestProcessTemplate
+	}
+	validationErrors := validateApplication(app)
+	if validationErrors != nil {
+		return dTypes.Application{}, errors.Join(validationErrors...)
+	}
+	return app, nil
+}
+
+func (c *CFProvider) createCFManifest(spaceName string) ([]cfTypes.AppManifest, error) {
+	ctx := context.Background()
+	log.Println("Analyzing space: ", spaceName)
+	spaceOpts := client.NewSpaceListOptions()
+	spaceOpts.Names.EqualTo(spaceName)
+	cfClient, err := c.GetClient()
+	if err != nil {
+		return nil, fmt.Errorf("error creating CF client: %v", err)
+	}
+	remoteSpace, err := cfClient.Spaces.First(ctx, spaceOpts)
+	if err != nil {
+		return nil, fmt.Errorf("error finding CF space for spaceName %s: %v", spaceName, err)
+	}
+	appsOpt := client.NewAppListOptions()
+	appsOpt.SpaceGUIDs.EqualTo(remoteSpace.GUID)
+	apps, err := cfClient.Applications.ListAll(context.Background(), appsOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error listing CF apps for space %s: %v", spaceName, err)
+	}
+	log.Println("Apps discovered: ", len(apps))
+
+	appManifests := []cfTypes.AppManifest{}
+	for _, app := range apps {
+		log.Println("Processing app:", app.Name)
+		appEnv, err := cfClient.Applications.GetEnvironment(context.Background(), app.GUID)
+		if err != nil {
+			return nil, fmt.Errorf("error getting environment for app %s: %v", app.GUID, err)
+		}
+		log.Printf("App Environment Variables: %s", appEnv)
```

Comment:
be careful about exposing the env values because they can have sensitive information